internal/controllers/variable_source.go

```diff
@@ -17,26 +17,68 @@ limitations under the License.
 package controllers
 
 import (
+	"context"
+
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	"github.com/operator-framework/deppy/pkg/deppy"
 	"github.com/operator-framework/deppy/pkg/deppy/input"
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
 
+	operatorsv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
 	"github.com/operator-framework/operator-controller/internal/resolution/variablesources"
 )
 
-func NewVariableSource(cl client.Client, catalogClient variablesources.BundleProvider) variablesources.NestedVariableSource {
-	return variablesources.NestedVariableSource{
+// BundleProvider provides the way to retrieve a list of Bundles from a source,
+// generally from a catalog client of some kind.
+type BundleProvider interface {
```

Comment:
Are there multiple options in use? Why do we need this interface?

Reply:
@stevekuznetsov at the moment we have two options: in-cluster one (reads from catalogd HTTP server) and offline one for resolution CLI PoC.